homeassistant/components/husqvarna_automower/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow to add the integration via the UI."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import data_entry_flow
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HusqvarnaConfigFlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler,
+    data_entry_flow.FlowHandler,
+    domain=DOMAIN,
+):
+    """Handle a config flow."""
+
+    DOMAIN = DOMAIN
+    VERSION = 1
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create an entry for the flow."""
+        return await self.async_step_finish(DOMAIN, data)
+
+    async def async_step_finish(self, unique_id, data) -> FlowResult:
```

Comment:
You can merge this step with the step above as there is no real reason to split it out